snuba/subscriptions/data.py

```diff
@@ -129,6 +141,57 @@ class LegacySubscriptionData(SubscriptionData):
 
     conditions: Sequence[Condition]
     aggregations: Sequence[Aggregation]
+    organization: Optional[int] = None
+    limit: int = 1
+    offset: int = 0
+    granularity: int = 3600
+
+    @staticmethod
+    def has_crash_rate_alert_aggregation(aggregations: Sequence[Aggregation]) -> bool:
+        for aggregation in aggregations:
+            try:
+                if aggregation[2] and bool(
+                    CRASH_RATE_ALERT_PATTERN.match(aggregation[2])
+                ):
+                    return True
+            except IndexError:
+                continue
+        return False
+
+    @classmethod
+    def update_sessions_subscriptions_with_session_params(
+        cls, data_dict: Dict[str, Any], data: Mapping[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Function that checks if this instance of LegacySubscriptionData is for a Crash Rate Alert
+        by checking if there is a `crash_free_percentage` aggregation, and if so updates the data
+        dictionary used to create an instance from this class.
+        """
+        if cls.has_crash_rate_alert_aggregation(data.get("aggregations", [])):
+            if not data.get("organization"):
+                raise InvalidQueryException(
+                    "No organization provided for Sessions subscription"
+                )
+
+            # Querying over the sessions dataset requires adding the following required fields:
+            # organization, offset and limit.
```

Comment:
Why do you say offset is a required field in sessions? I don't think it's there at all.

Reply:
I mistook an error message